### PR TITLE
First homework task-FizzBuzz

### PR DESCRIPTION
**Pull request by Rzucky**

### CODE REVIEW

File: homework_1/task1/fizzbuzz/fizzBuzz.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		return nil, errors.New("range start is greater than range end")
+	}
+
+	if start <= 0 {
+		return nil, errors.New("start is 0 or negative")
+	}
+
+	if end <= 0 {
+		return nil, errors.New("end is 0 or negative")
+	}
```
Comment (oreskdaxilis) [inline homework_1/task1/fizzbuzz/fizzBuzz.go:20]: Considering the first 2 checks, this check is redundant (it's not possible for end to be negative).

File: homework_1/task1/fizzbuzz/fizzBuzz.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	var changedOutput []string
+
+	for i := start; i <= end; i++ {
+		switch{
```
Comment (oreskdaxilis) [inline homework_1/task1/fizzbuzz/fizzBuzz.go:25]: Should be a whitespace before `{` always except when creating a new struct.
You can automatically format your code with `go fmt ./...` from `homework_2` directory and it should format all .go files in that directory and its child directories.

File: homework_1/task1/fizzbuzz/fizzBuzz.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		//using mod 15 can save one check instead of (mod 3 && mod 5)
```
Comment (oreskdaxilis) [inline homework_1/task1/fizzbuzz/fizzBuzz.go:26]: Should be a whitespace after `//`, but formatting should resolve that issue as well.

File: homework_1/task1/fizzbuzz/testCases_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+type testCase struct {
+	inputStart   int
+	inputEnd     int
+
+	expectedOutput []string
+	expectingError bool
+}
+
+
+func getTestCases() []testCase {
```
Comment (oreskdaxilis) [inline homework_1/task1/fizzbuzz/testCases_test.go:12]: Nice list of test cases. 👍

File: homework_1/task1/fizzbuzz/fizzbuzz_test.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	for idx, x := range first {
+		if x != second[idx] {
+			return false
+		}
+	}
+
+	return true
+}
```
Comment (oreskdaxilis) [inline homework_1/task1/fizzbuzz/fizzbuzz_test.go:21]: Even though this function is entirely correct, you don't need it for testing purposes - you can validate two slices are equal by calling `So(actual, ShouldResemble, expected)` from `goconvey` library - e.g. https://github.com/superbet-group/code-cadets-2021/blob/main/lecture_1/07_testing/filter/divisorFilterConvey_test.go

File: homework_1/task1/fizzbuzz/fizzBuzz.go (hunk 2 above)
Comment (Rzucky) [reply homework_1/task1/fizzbuzz/fizzBuzz.go:25]: Neat trick, thank you! I was doing it by hand 😄

File: homework_1/task1/fizzbuzz/fizzBuzz.go (hunk 3 above)
Comment (Rzucky) [reply homework_1/task1/fizzbuzz/fizzBuzz.go:26]: `go fmt ./...`  actually didn't. Maybe i have to tweak it somewhere in Settings. But I've done it by hand.

File: homework_1/task1/fizzbuzz/testCases_test.go (hunk 4 above)
Comment (Rzucky) [reply homework_1/task1/fizzbuzz/testCases_test.go:12]: Thank you! 👍

File: homework_1/task1/fizzbuzz/fizzbuzz_test.go (hunk 5 above)
Comment (Rzucky) [reply homework_1/task1/fizzbuzz/fizzbuzz_test.go:21]: Yea, we did two versions, for some reason i chose this one 😄

**Comment by Rzucky** [comment]
I'll close this branch since the other one has these files, but the updates were done.

**Pull request closed without merging**